client/flaschen.py

```diff
@@ -0,0 +1,64 @@
+# -*- mode: c++; c-basic-offset: 4; indent-tabs-mode: nil; -*-
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation version 2.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://gnu.org/licenses/gpl-2.0.txt>
+
+import socket
+
+class Flaschen(object):
+  '''A Framebuffer display interface that sends a frame via UDP.'''
+
+  def __init__(self, host, port, width, height, layer=0):
+    self.host = host
+    self.port = port
+    self.width = width
+    self.height = height
+    self.layer = layer
+    self.pixels = []
+    for x in xrange(width):
+      self.pixels.append([(0, 0, 0) for y in xrange(height)])
+    self.sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+
+  def _header(self):
+    return ''.join(["P6\n",
+                    "45 35\n",
+                    "255\n"])
+
+  def _footer(self):
+    return ''.join(["0\n",
+                    "0\n",
+                    "%d\n" % self.layer])
+
+  def set(self, x, y, color, transparent=False):
```

Comment:
One API level decision might be: should 'transparent' be a parameter of 'set()' or rather be a global property set in the constructor ?
Intuitively, I'd expect it to be more useful in the constructor - because if someone wanted to do it selectively in the set call, they could do the Python equivalent of `c == (0, 0, 0) ? (1, 1, 1) : c` anyway; but the decision to do something transparently might typically at the time we decide on which layer to go.
Your call.